monticore-grammar/src/test/java/de/monticore/types/check/SynthesizeComponentFromMCBasicTypesTest.java

```diff
@@ -0,0 +1,162 @@
+import de.monticore.types.mcbasictypes._ast.ASTMCVoidType;
+import de.se_rwth.commons.logging.Log;
+import de.se_rwth.commons.logging.LogStub;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+public class SynthesizeComponentFromMCBasicTypesTest extends AbstractMCTest {
+
+  @BeforeEach
+  public void setup() {
+    LogStub.init();
+    Log.enableFailQuick(false);
+    Log.clearFindings();
+
+    ComponentSymbolsWithMCBasicTypesTestMill.reset();
+    ComponentSymbolsWithMCBasicTypesTestMill.init();
+  }
+
+  @Test
+  public void shouldHandleMCQualifiedType() {
+    var globalScope = ComponentSymbolsWithMCBasicTypesTestMill.globalScope();
+
+    // Case 1: simple name "Foo"
+    String simpleName = "Foo";
+    ASTMCQualifiedType astFoo = MCTypeFacade.getInstance().createQualifiedType(simpleName);
+    astFoo.setEnclosingScope(globalScope);
+
+    ComponentTypeSymbol fooSymbol = ComponentSymbolsWithMCBasicTypesTestMill.componentTypeSymbolBuilder()
+        .setName(astFoo.getMCQualifiedName().getBaseName())
+        .setSpannedScope(ComponentSymbolsWithMCBasicTypesTestMill.scope())
+        .build();
+
+    globalScope.add(fooSymbol);
+    globalScope.addSubScope(fooSymbol.getSpannedScope());
+
+    CompKindCheckResult resultFoo = new CompKindCheckResult();
+    SynthesizeCompKindFromMCBasicTypes synthFoo =
+      new SynthesizeCompKindFromMCBasicTypes(resultFoo);
+
+    // When
+    synthFoo.handle(astFoo);
+
+    // Then
+    Assertions.assertTrue(resultFoo.getResult().isPresent());
+    Assertions.assertTrue(resultFoo.getResult().get().isComponentType());
+    Assertions.assertEquals(fooSymbol, resultFoo.getResult().get().getTypeInfo());
+
+    // Case 2: qualified name "qual.Foo"
+    String qualifiedName = "qual.Foo";
+    ASTMCQualifiedType astQualFoo = MCTypeFacade.getInstance().createQualifiedType(qualifiedName);
+    astQualFoo.setEnclosingScope(globalScope);
+
+    ComponentTypeSymbol qualFooSymbol =
+      ComponentSymbolsWithMCBasicTypesTestMill.componentTypeSymbolBuilder()
+        .setName(astQualFoo.getMCQualifiedName().getBaseName()) // "Foo"
+        .setSpannedScope(ComponentSymbolsWithMCBasicTypesTestMill.scope())
+        .build();
+
+    // Build qualifier scope "qual" and put Foo inside it
+    var qualScope = ComponentSymbolsWithMCBasicTypesTestMill.scope();
+    qualScope.setName("qual");
+    qualScope.add(qualFooSymbol);
+    qualScope.addSubScope(qualFooSymbol.getSpannedScope());
+    globalScope.addSubScope(qualScope);
+
+    CompKindCheckResult resultQualFoo = new CompKindCheckResult();
+    SynthesizeCompKindFromMCBasicTypes synthQualFoo =
+      new SynthesizeCompKindFromMCBasicTypes(resultQualFoo);
+
+    // When
+    synthQualFoo.handle(astQualFoo);
+
+    // Then
+    Assertions.assertTrue(resultQualFoo.getResult().isPresent());
```

Comment:
Das ist schon besser, aber hier werden immer noch 2 Sachen getestet in einem Test. Bau das sonst in eine zweite Methode, wenn du das nicht gut parametrisieren kannst.